### PR TITLE
Added comments for cmake test scripts.

### PR DESCRIPTION
**Pull request by nmoinvaz**
I have added comments to these scripts I created. Hopefully these make it easy to follow.

### CODE REVIEW

File: cmake/test-compress.cmake [hunk 1]
```diff
@@ -1,3 +1,28 @@
...
+
+# Required Variables
+#   INPUT                   - Input file to test
+#   TARGET or               - Command to run for both compress and decompress
+#     COMPRESS_TARGET and   - Command to run to compress input file
+#     DECOMPRESS_TARGET     - Command to run to decompress output file
+
+# Optional Variables
+#   COMPRESS_ARGS           - Arguments to pass for compress command (default: -c -k)
+#   DECOMPRESS_ARGS         - Arguments to pass to decompress command (default: -d -c)
```
Comment (Dead2) [inline cmake/test-compress.cmake:19]: Why are these sent as semicolon-separated lists from CMakeLists.txt?
Ex line 1218: `"-DCOMPRESS_ARGS=-c;-k;-4"`

Comment (nmoinvaz) [reply cmake/test-compress.cmake:19]: That is the equivalent of a list in CMake or at least how it stores them.

Comment (nmoinvaz) [reply cmake/test-compress.cmake:19]: It is better for it to be a list in the event that it tries to double quote each argument when calling the command. "-c" "-k" versus "-c -k".

Comment (Dead2) [reply cmake/test-compress.cmake:19]: Ah, that makes sense then. Thanks for the explanation. 👍

**Pull request merged**